src/status_im/navigation/events.cljs

```diff
@@ -74,6 +74,12 @@
   [{:keys [db]} root-id]
   {:set-root [root-id (:theme db)]})
 
+(rf/defn update-theme-and-init-root
+  {:events [:update-theme-and-init-root]}
+  [_ root-id]
+  {:fx [[:dispatch [:theme/switch {:view-id root-id}]]
```

Comment:
Looks like a proper solution to use ordering from `:fx` 👍🏼, but haven't tested to be sure the fix is good.